discovery-infra/tests/test_validations.py

```diff
@@ -114,3 +122,89 @@ def test_hosts_and_cluster_max_length(self, cluster, nodes):
         self.assert_string_length(long_hostname, 63)
         c.set_host_name(host_id=random_master_host["id"], requested_name=long_hostname)
         c.start_install_and_wait_for_installed()
+
+    @pytest.mark.regression
+    def test_cluster_nodes_count(self, nodes, cluster):
+        # This test check that cluster has exactly 3 master nodes, or 3 masters and more then 1 workers.
+
+        # Define new cluster
+        new_cluster = cluster()
+        # Wait for cluster to be in Ready to install (with 3 masters and 2 workers)
+        new_cluster.prepare_for_install(nodes=nodes)
+        cluster_details = new_cluster.get_details()
+        self.assert_cluster_validation(cluster_details, "hosts-data", "sufficient-masters-count", "success")
+        # set master role to a worker host
+        random_worker = new_cluster.get_random_host_by_role(role=consts.NodeRoles.WORKER)
+        new_cluster.set_specific_host_role(random_worker, role=consts.NodeRoles.MASTER)
+        new_cluster.wait_for_cluster_validation("hosts-data", "sufficient-masters-count", ["failure"])
+        # set back worker role
+        new_cluster.set_specific_host_role(random_worker, role=consts.NodeRoles.WORKER)
+        new_cluster.wait_for_cluster_validation("hosts-data", "sufficient-masters-count", ["success"])
+        # Disable the first worker node
+        worker_hosts = new_cluster.get_hosts_by_role(role=consts.NodeRoles.WORKER)
+        worker_host1 = worker_hosts[0]
+        worker_host2 = worker_hosts[1]
+        new_cluster.disable_host(host=worker_host1)
+        cluster_details = new_cluster.get_details()
+        self.assert_cluster_validation(cluster_details, "hosts-data", "sufficient-masters-count", "failure")
+        # Disable the second worker node
+        new_cluster.disable_host(host=worker_host2)
+        cluster_details = new_cluster.get_details()
+        new_cluster.wait_for_ready_to_install()
```

Comment:
Why do you need to wait_for_ready_to_install? If need then you should new_cluster.get_details() after the wait, no?

Reply:
Your right, its redundant, removed

Reply:
In second though, no - it should be there. Waiting for cluster validation to pass after disable node

Reply:
when only one node disable - cluster validation fail.
when 2 worker are disabled, than cluster validation should pass, and cluster should turn to ready to install

Reply:
Okay,
Still shouldn't you new_cluster.get_details() after new_cluster.wait_for_ready_to_install()?

Reply:
Yep, fixed